core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -213,10 +217,19 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
     if (leaderId.isEmpty)
       leaderId = Some(member.memberId)
     members.put(member.memberId, member)
+    member.supportedProtocols.foreach{ case (protocol, _) => supportedProtocols(protocol) += 1 }
+    member.awaitingJoinCallback = callback
+    if (member.awaitingJoinCallback != null)
+      awaitingJoinCallbackMembers += 1;
   }
 
   def remove(memberId: String) {
-    members.remove(memberId)
+    val member = members.remove(memberId)
+    if (member.nonEmpty) {
```

Comment:
use match instead of if? why?

Reply:
`Option.get` usage is discouraged since it's unsafe. Instead one should either pattern match or use `foreach` and so on.

Here. you can write the code as:

```scala
members.remove(memberId).foreach { member =>
...
}
```

Reply:
got it! @ijuma thank you for the explanation. I will fix this